test/test_transformers.py

```diff
@@ -724,6 +731,58 @@ def rand_tensor(*shape):
             if dropout_p == 0.0 or device == 'cpu':
                 self.assertEqual(actual, expected)
 
+    @unittest.skipIf(TEST_WITH_CROSSREF, 'Fastpath not available with crossref')
+    def test_mask_check_fastpath(self):
+        """
+        Test that fastpath is executed independently of the mask that is passed.
+        If the passed mask is left aligned, test that nested tensors are used (sparsity fastpath),
+        otherwise use fastpath with traditional tensors.
+        """
+
+        with torch.no_grad():
+            x = torch.Tensor([[[1, 2], [3, 4], [5, 6]]]).to(torch.float)
+
+            def _test_fastpath(model, mask, mock_return_value, nested_tensors=True):
+                with patch('torch._transformer_encoder_layer_fwd') as fastpath_mock:
+                    fastpath_mock.return_value = mock_return_value
+                    model(x, src_key_padding_mask=mask)
+                    self.assertTrue(fastpath_mock.called)
```

Comment:
don't we expect this to be False for certain inputs?

Reply:
Yes, but not in the context of the `mask_check` flag which is what we are testing right now. There are multiple cases where fastpath is not taken (like using grad, or model in training mode) but right now we are assuming those conditions are met to test just that flag.

Reply:
Cool, I'm good with this